components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java

```diff
@@ -192,6 +192,20 @@ protected void activate(ComponentContext ctxt) {
             builder.setBundleContext(ctxt.getBundleContext());
             builder.buildEntitlementConfig(EntitlementConfigHolder.getInstance());
 
+            boolean balanaConfig = Boolean.parseBoolean((String) EntitlementServiceComponent.getEntitlementConfig().
+                    getEngineProperties().get(PDPConstants.BALANA_CONFIG_ENABLE));
+            
+            String configProperty = System.getProperty("org.wso2.balana.PDPConfigFile");
+
+            if (balanaConfig && configProperty == null) {
+                String configFilePath = CarbonUtils.getCarbonConfigDirPath() + File.separator + "security" 
+                        + File.separator + "balana-config.xml";
+                
+                System.setProperty("org.wso2.balana.PDPConfigFile", configFilePath);
+                
+                log.debug("Setting org.wso2.balana.PDPConfigFile property to " + configFilePath);
+            }
+
```

Comment:
Let's move the debug log after the If condition also let's have the guard,

```
if (log.isDebugEnabled()) {
log.debug("Setting org.wso2.balana.PDPConfigFile property to " + configFilePath);
}
```

Reply:
@mefarazath fixed